Content.Server/Storage/EntitySystems/StorageSystem.cs

```diff
@@ -261,32 +260,33 @@ private void HandleEntityInsertedIntoContainer(EntInsertedIntoContainerMessage m
 
         private void CheckSubscribedEntities(ServerStorageComponent storageComp)
         {
+            var xform = Transform(storageComp.Owner);
+            var storagePos = xform.WorldPosition;
+            var storageMap = xform.MapID;
 
-            // We have to cache the set of sessions because Unsubscribe modifies the original.
-            _sessionCache.Clear();
-            _sessionCache.AddRange(storageComp.SubscribedSessions);
-
-            if (_sessionCache.Count == 0)
-                return;
-
-            var storagePos = EntityManager.GetComponent<TransformComponent>(storageComp.Owner).WorldPosition;
-            var storageMap = EntityManager.GetComponent<TransformComponent>(storageComp.Owner).MapID;
+            var remove = new RemQueue<IPlayerSession>();
 
-            foreach (var session in _sessionCache)
+            foreach (var session in storageComp.SubscribedSessions)
             {
                 // The component manages the set of sessions, so this invalid session should be removed soon.
                 if (session.AttachedEntity is not {} attachedEntity || !EntityManager.EntityExists(attachedEntity))
                     continue;
 
-                if (storageMap != EntityManager.GetComponent<TransformComponent>(attachedEntity).MapID)
+                var attachedXform = Transform(attachedEntity);
+                if (storageMap != attachedXform.MapID)
                     continue;
 
-                var distanceSquared = (storagePos - EntityManager.GetComponent<TransformComponent>(attachedEntity).WorldPosition).LengthSquared;
+                var distanceSquared = (storagePos - attachedXform.WorldPosition).LengthSquared;
                 if (distanceSquared > SharedInteractionSystem.InteractionRangeSquared)
                 {
-                    storageComp.UnsubscribeSession(session);
+                    remove.Add(session);
                 }
             }
+
+            foreach (var session in remove)
+            {
+                storageComp.SubscribedSessions.Remove(session);
```

Comment:
This should be  `UnsubscribeSession()`, right?

Reply:
yep